athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/propertygraph/PropertyGraphHandler.java

```diff
@@ -0,0 +1,231 @@
+     *                           Split 2. The Catalog, Database, and Table the read
+     *                           request is for. 3. The filtering predicate (if any)
+     *                           4. The columns required for projection.
+     * @param queryStatusChecker A QueryStatusChecker that you can use to stop doing
+     *                           work for a query that has already terminated
+     * @throws Exception
+     * @note Avoid writing >10 rows per-call to BlockSpiller.writeRow(...) because
+     *       this will limit the BlockSpiller's ability to control Block size. The
+     *       resulting increase in Block size may cause failures and reduced
+     *       performance.
+     */
+
+    public void executeQuery(ReadRecordsRequest recordsRequest, final QueryStatusChecker queryStatusChecker,
+            final BlockSpiller spiller) throws Exception 
+    {
+        logger.info("readWithConstraint: enter - " + recordsRequest.getSplit());
+        long numRows = 0;
+        Client client = neptuneConnection.getNeptuneClientConnection();
+        GraphTraversalSource graphTraversalSource = neptuneConnection.getTraversalSource(client);
+        GraphTraversal graphTraversal = null;
+        TableName tableName = recordsRequest.getTableName();
+        String labelName = tableName.getTableName();
+        String tableSchemaMetaType = recordsRequest.getSchema().getCustomMetadata().get("type");
+        GeneratedRowWriter.RowWriterBuilder builder = GeneratedRowWriter.newBuilder(recordsRequest.getConstraints());
+        
+        logger.info("readWithConstraint: schema type is " + tableSchemaMetaType);
+
+        if (tableSchemaMetaType != null) {
+            switch (tableSchemaMetaType) {
+                case "vertex":
+                    graphTraversal = graphTraversalSource.V().hasLabel(labelName);
+                    getQueryPartForContraintsMap(graphTraversal, recordsRequest);
+                    graphTraversal = graphTraversal.valueMap().with(WithOptions.tokens);
+
+                    for (final Field nextField : recordsRequest.getSchema().getFields()) {
+                        VertexRowWriter.writeRowTemplate(builder, nextField);
+                    }
+
+                    break;
+
+                case "edge":
+                    graphTraversal = graphTraversalSource.E().hasLabel(labelName);
+                    getQueryPartForContraintsMap(graphTraversal, recordsRequest);
+                    graphTraversal = graphTraversal.elementMap();
+
+                    for (final Field nextField : recordsRequest.getSchema().getFields()) {
+                        EdgeRowWriter.writeRowTemplate(builder, nextField);
+                    }
+
+                    break;
+            }
+        }
+
+        // log string equivalent of gremlin query
+        logger.info("readWithConstraint: enter - "
+                + GroovyTranslator.of("g").translate(graphTraversal.asAdmin().getBytecode()));
```

Comment:
should we move this to DEBUG? since it might result in logging sensitive info?

Reply:
done